### ISSUE TITLE
Very slow performance with 'Svelte for VS Code' extension on Sapper projects on networked drives

### ISSUE THREAD

**Issue by Runi-c** [sveltejs/language-tools#933]
After trying the extension on a fresh project & seeing how fast it is, I'd like to reach out for some assistance in figuring out why the language server takes **6-7 seconds** on every edit to HTML to begin serving language features again on Sapper projects on a networked drive.

Setting debug trace to "verbose" produces this series of messages when typing the below snippet and expecting IDE features:

https://user-images.githubusercontent.com/5150427/113370742-e16aaa80-9321-11eb-9830-29eaf9058b4e.mp4

```
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("<").
[Trace - 7:19:58 PM] Sending request 'textDocument/completion - (5)'.
[Trace - 7:19:58 PM] Sending request 'textDocument/signatureHelp - (6)'.
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("B").
[Trace - 7:19:58 PM] Sending notification '$/cancelRequest' - (6).
[Trace - 7:19:58 PM] Sending request 'textDocument/signatureHelp - (7)'.
[Trace - 7:19:58 PM] Sending request 'textDocument/semanticTokens/full - (8)'.
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("u").
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("t").
[Trace - 7:19:58 PM] Sending notification '$/cancelRequest' - (7).
[Trace - 7:19:58 PM] Sending request 'textDocument/signatureHelp - (9)'.
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("t").
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("t").
[Trace - 7:19:58 PM] Sending notification '$/cancelRequest' - (9).
[Trace - 7:19:58 PM] Sending request 'textDocument/signatureHelp - (10)'.
[Trace - 7:19:58 PM] Sending notification 'textDocument/didChange' ("o").
[Trace - 7:19:59 PM] Sending notification '$/cancelRequest' - (10).
[Trace - 7:19:59 PM] Sending request 'textDocument/signatureHelp - (11)'.
[Trace - 7:19:59 PM] Sending request 'textDocument/documentColor - (12)'.
[Trace - 7:19:59 PM] Sending request 'textDocument/codeAction - (13)'.
[Trace - 7:20:02 PM] Sending notification '$/cancelRequest' - (5).
[Trace - 7:20:03 PM] Received response 'textDocument/completion - (5)' in 5839ms.
Using Svelte v3.32.3 from ~\Frontend\node_modules\svelte\compiler
Using Svelte v3.32.3 from ~\Frontend\node_modules\svelte\compiler
[Trace - 7:20:03 PM] Received notification 'textDocument/publishDiagnostics'.
[Trace - 7:20:03 PM] Received response 'textDocument/signatureHelp - (6)' in 5775ms.
[Trace - 7:20:04 PM] Sending request 'textDocument/codeAction - (14)'.
[Trace - 7:20:04 PM] Sending notification '$/cancelRequest' - (13).
[Trace - 7:20:09 PM] Received response 'textDocument/signatureHelp - (7)' in 11269ms.
Using Svelte v3.32.3 from ~\Frontend\node_modules\svelte\compiler
Using Svelte v3.32.3 from ~\Frontend\node_modules\svelte\compiler
[Trace - 7:20:09 PM] Received notification 'textDocument/publishDiagnostics'.
[Trace - 7:20:09 PM] Received response 'textDocument/semanticTokens/full - (8)' in 11304ms.
[Trace - 7:20:09 PM] Received response 'textDocument/signatureHelp - (9)' in 11076ms.
[Trace - 7:20:09 PM] Received response 'textDocument/signatureHelp - (10)' in 10879ms.
[Trace - 7:20:09 PM] Received response 'textDocument/signatureHelp - (11)' in 10716ms.
[Trace - 7:20:09 PM] Received response 'textDocument/documentColor - (12)' in 10669ms.
[Trace - 7:20:09 PM] Received response 'textDocument/codeAction - (13)' in 10594ms.
[Trace - 7:20:09 PM] Received response 'textDocument/codeAction - (14)' in 5515ms.
```

Trying the clean [Svelte template](https://github.com/sveltejs/template) and [Sapper webpack template](https://github.com/sveltejs/sapper-template) in different environments with trace on yields these results:
```
Local drive - Svelte template: 22ms
Network drive - Svelte template: 34ms
Local drive - Sapper Webpack: 307ms
Network Drive - Sapper Webpack: 6002ms
```
There didn't seem to be much difference with/without Typescript or with/without sveltestrap.

**Comment by dummdidumm** [comment]
Could you provide more info on how to reproduce this? Ideally the repo itself. If not, some steps on how to replicate it, like the contents of your `package.json` and `svelte.config.js` (if you have one), if you are using TS or not, if yes your `tsconfig.json`, how many files you project has, if it happens in all files or only some, etc etc.

**Comment by Runi-c** [comment]
The specific repo in question is private, but as detailed above, the stock Sapper Webpack template took 6002ms for IDE features to start working again. It's wholly bearable on a local drive, but on a networked drive it's unbearable.

After a bit more testing, it seems to be specifically the Sapper Webpack template, after the Typescript conversion script is ran, and `npm install` is ran. Before any of those things, the time taken is a tiny fraction of what I'm experiencing, and if the Typescript script isn't ran, the time is also extremely fast.

Therefore, here's the repro steps:
1. Open a command prompt in a networked location (not hosted locally).
2. `npx degit "sveltejs/sapper-template#webpack" my-app`
3. `cd my-app`
4. `npm i`
5. open the my-app folder in VSC with the Svelte for VS Code extension
6. open src/components/Nav.svelte, go to the bottom, and experiment with typing a `<`, `<a`, `<d`, etc. and see how long the extension takes to re-check the file and provide suggestions

**Comment by Runi-c** [comment]
Given the fact that the performance only deteriorates this excessively after the `npm install`, I went and tried removing the packages added by the typescript setup script one by one. The offending package seems to be `svelte-preprocess`. Why would the addition of this package cause the language server's performance to deteriorate by 5x?

**Comment by dummdidumm** [comment]
Oh I have an idea, thanks for investigating. I'll check when I have time.

**Comment by dummdidumm** [comment]
I just tried to reproduce this and for me the response is almost immediate. Are there other extensions you are using? What happens if you disable them? Which version of Svelte for VS Code is installed? Does the slowness appear after a fresh reload of VS Code or only after some time?

**Comment by PuruVJ** [comment]
Has this been fixed? My IDE went from instataneous perf in the starter vite template to 4 seconds for intellisense to kick in after just 15 components 🥲

**Comment by dummdidumm** [comment]
Is your issue about networked, drives, too? If not, please open a separate issue with more details, ideally a reproduction repository.

**Comment by PuruVJ** [comment]
Done @dummdidumm https://github.com/sveltejs/language-tools/issues/1139 👍

**Comment by dummdidumm** [comment]
The latest version of `svelte-preprocess` does cache the resolved tsconfig files. Previously, they were loaded for each file, which might be slow on networked drives because IO operations could be slower. Could you try and report back if the situation improved?

**Comment by Runi-c** [comment]
Apologies for the long delay. There does seem to be a considerable improvement. It's still slow, but I understand there's only so much that can be done about that over a networked drive and for this reason I've switched to remote windows with VSCode which has vastly improved my development experience. Svelte language features are quite snappy over the remote window.

I think I can consider this solved though, the delay now seems to be about 2 seconds instead of 6-7+. Thanks for your efforts!

**Comment by rafaelspring** [comment]
@Roryl-c  I am also seeing very slow plugin performance for large-ish svelte files (> 1000 LOC).

> Setting debug trace to "verbose" produces this series of messages when typing the below snippet and expecting IDE features:

Can you let me know how to do this? I tried googling "vs code debug trace verbose" but no good results came up.

**Comment by Ayfri** [comment]
Any update? The issue is still occurring!
